### DISCUSSION TITLE
can tud_audio_fb_set() be called from inside tud_audio_feedback_interval_isr()?

### DISCUSSION THREAD

**Discussion by mastupristi** [hathach/tinyusb#2498] [Q&A]
I have this curiosity:
can `tud_audio_fb_set()` be called from inside an interrupt handler such as `tud_audio_feedback_interval_isr()`?
I think the question is legit, because calling `tud_audio_fb_set()` causes the use of FreeRTOS functions that should not be called from inside an interrupt.

regards
Max

**Comment by HiFiPhile** [comment]
You can even run the whole TUSB stack inside IRQ, it's what I does to ensure audio streaming has the highest priority.

But this usage doesn't support RTOS yet, so you need to set  `CFG_TUSB_OS = OPT_OS_NONE` and take care of concurrency manually.

```
void USB1_IRQHandler(void)
{
  tud_int_handler(1);
  tud_task();
}
```
